geospaas_rest_api/tests/test_read_only_api.py

```diff
@@ -165,151 +157,177 @@ class DatasetFilteringTests(django.test.TestCase):
+            '&geographic_location__geometry__contains=POINT+%289+9%29'
+            '&source__instrument__short_name__contains=HXT'
+        )
+        self.assertJSONEqual(response.content, {
+            'next': None, 'previous': None, 'results': [self.DATASET_DICT_1]}
+        )
+
+    def test_invalid_direct_lookup(self):
+        """An error must happen when an invalid lookup is submitted
+        on a "direct" (i.e. not related) filter
+        """
+        response = self.client.get('/api/datasets/?entry_id__contai=titusen')
+        self.assertEqual(response.status_code, 400)
+        self.assertJSONEqual(response.content, ["Invalid lookups found: ['entry_id__contai']"])
 
-    def test_filter_with_location(self):
-        """The location parameter is present in the request"""
-        self.request_mock.query_params = {
-            self.filter.LOCATION_PARAM: 'test_location'
-        }
-        self.filter.to_html(self.request_mock, self.queryset, self.view_mock)
-        self.mock_render.assert_called_with({
-            'params': self.params_list,
-            'terms': ['', 'test_location', '']
-        })
 
-    def test_filter_with_source(self):
-        """The source parameter is present in the request"""
-        self.request_mock.query_params = {
-            self.filter.SOURCE_PARAM: 'test_source'
-        }
-        self.filter.to_html(self.request_mock, self.queryset, self.view_mock)
-        self.mock_render.assert_called_with({
-            'params': self.params_list,
-            'terms': ['', '', 'test_source']
-        })
+    def test_invalid_related_lookup(self):
+        """An error must happen when an invalid lookup is submitted
+        on a related filter
+        """
+        response = self.client.get('/api/datasets/?source__instrument__short_name__foo=HXT')
+        self.assertEqual(response.status_code, 400)
+        self.assertJSONEqual(response.content, [
+            "Invalid lookups found: ['source__instrument__short_name__foo']"
+        ])
 
+    def test_invalid_lookup_multiple_parameters(self):
+        """An error must happen when an invalid lookup is submitted
+        on a filter, even when other parameters are present
+        """
+        response = self.client.get(
+            '/api/datasets/?entry_id__contai=titusen'
+            '&summary__contains=test'
+        )
+        self.assertEqual(response.status_code, 400)
+        self.assertJSONEqual(response.content, ["Invalid lookups found: ['entry_id__contai']"])
 
-class DatasetURIFilteringTests(django.test.TestCase):
+    def test_invalid_exact_lookup(self):
+        """An error must happen when an invalid "exact" lookup
+        is submitted
+        """
+        response = self.client.get('/api/datasets/?sourc=1')
+        self.assertEqual(response.status_code, 400)
+        self.assertJSONEqual(response.content, [
+            "Invalid lookups found: ['sourc']"
+        ])
+
+
+class DatasetURIFilterFilteringTests(django.test.TestCase):
```

Comment:
`FilterFiltering`? please simplify the name

Reply:
done